### DISCUSSION TITLE
htmx throws exception on certain  hx-trigger="sse:event/names"

### DISCUSSION THREAD

**Discussion by hspaay** [bigskysoftware/htmx#2731] [Q&A] [answered]
In a small iot project I've got a golang template page that uses sse for refreshing a section of the page.
The problem is that the event names can contain spaces and '/' characters, which is valid in sse. htmx:sse throws an exception when using these characters.

For An iot device 'device1' with the property name 'prop of 1' the sse event name is "device1/prop of 1"
A snippet of the template:
```<article
   hx-trigger="sse:{{.DeviceName}}/{{.PropName}}"
   hx-get="/iotdata/{{.DeviceName}}/{{.PropName}}"
   hx-target="#targetElement"
>
```

This works fine for hx-get. Even sse-swap works like a charm. hx-trigger however doesn't accept the '/' and ' ' characters.

I read the htmx specification but could not locate any restrictions on the use of event names. Nor does hx-trigger mention any, although obviously it has a use for spaces a separator.

Is there a way to escape the event name for hx-trigger="sse:" in the template?

**Comment by Telroshan** [comment, marked as answer]
Hey, unfortunately there's no support for event names containing spaces for now in `hx-trigger`. Spaces are used as separators as you mentioned, to process [modifiers](https://htmx.org/attributes/hx-trigger/).
In this case you will probably want to simply send events that don't contain any space from your backend.
Relevant line of code in the lib btw:
https://github.com/bigskysoftware/htmx/blob/1c1e4faab953700e2147e3115013a56b542661ed/src/htmx.js#L2120

**Reply by hspaay** [reply]
Thanks for confirming @Telroshan and explaining the reason. Yes it looks like a 'no-spaces' requirement must be enforced on the backend. 
Cheers,
